### ISSUE TITLE
False positives in rule ID 942430

### ISSUE THREAD

**Issue by shieku** [coreruleset/coreruleset#3284]
### Description

Hello
I use CRS 3.2 with Web Application Firewall on Azure.
Rule ID942430 was detected when sending a request at the following URL:
```
https://xxx/yyy?zzz=%E6%98%8E%E7%B4%B0%E6%98%8E%E7%B4%B0%E6%98%8E%E7%B4%B0%E6%98%8E%E7%B4%B0%E6%98%8E%E7%B4%B0%E6%98%8E%E7%B4%B0%E6%98%8E%E7%B4%B0%E6%98%8E%E7%B4%B0%E6%98%8E%E7%B4%B0%E6%98%8E%E7%B4%B0%E6%98%8E%E7%B4%B0%E6%98%8E%E7%B4%B0%E6%98%8E%E7%B4%B0%E6%98%8E%E7%B4%B0%E6%98%8E%E7%B4%B0%E6%98%8E%E7%B4%B0%E6%98%8E%E7%B4%B0%E6%98%8E%E7%B4%B0%E6%98%8E%E7%B4%B0
```
Decoding the query parameters results in the following URL:
```
https://xxx/yyy?zzz=明細明細明細明細明細明細明細明細明細明細明細明細明細明細明細明細明細明細明細
```
The following was recorded in the WAF log.
```
ruleId: 942430
Message: Restricted SQL Character Anomaly Detection (args): # of special characters exceeded (12)
details_data: {��細明細明細明細明細明�� found within [ARGS:DETAILS_TXT:明細明細明細明細明細明細明細明細明細明細明細明細明細明細明細明細明細明細明細]}
```
I tried entering some double-byte characters instead of the word "明細", but the same rule was not detected.
Why is this rule detected only for the above query parameters?
I want to know the cause of this.

Thank you and best regards,

**Comment by RedXanadu** [comment]
Hi @shieku,

Please use our false positive issue template in the future (provided by default). This ensures that we have all the required information in a standard format, making the process of triaging issues faster and simpler.

It looks like the regular expression that powers rule 942430 is matching against parts of the UTF-8 encoded characters in your sample input. That is what is causing your false positive.

CRS is known to struggle with issues like this with CJK characters, unfortunately (and non-Latin scripts in general). This is something we plan to work to improve in the future.

For now, you can resolve your issue by tuning away the false positive with a rule exclusion. As you're using Azure WAF, you will need to refer to the Azure-specific documentation to find out how to do that (the official CRS documentation will not be of much help, unfortunately.)

**Comment by theseion** [comment]
Closing this as it has likely been resolved. @shieku, please reopen if you need more help.
